org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/AllBreakpointsEnablementAction.java

```diff
@@ -0,0 +1,125 @@
+import org.eclipse.debug.internal.ui.actions.ActionMessages;
+import org.eclipse.debug.internal.ui.preferences.IDebugPreferenceConstants;
+import org.eclipse.jface.action.IAction;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.MessageDialogWithToggle;
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.ui.IWorkbenchWindow;
+
+abstract class AllBreakpointsEnablementAction extends AbstractRemoveAllActionDelegate implements IBreakpointsListener {
+
+	private final boolean fEnable;
+	private final String fLabel1;
+	private final String fLabel2;
+
+	AllBreakpointsEnablementAction(boolean enable, String label1, String label2) {
+		this.fEnable = enable;
+		this.fLabel1 = label1;
+		this.fLabel2 = label2;
+	}
+
+	@Override
+	protected boolean isEnabled() {
+		return DebugPlugin.getDefault().getBreakpointManager().getBreakpoints().length > 0;
+	}
+
+	@Override
+	public void breakpointsAdded(IBreakpoint[] breakpoints) {
+		update();
+	}
+
+	@Override
+	public void breakpointsChanged(IBreakpoint[] breakpoints, IMarkerDelta[] deltas) {
+		update();
+	}
+
+	@Override
+	public void breakpointsRemoved(IBreakpoint[] breakpoints, IMarkerDelta[] deltas) {
+		if (getAction() != null) {
+			update();
+		}
+	}
+
+	@Override
+	protected void initialize() {
+		DebugPlugin.getDefault().getBreakpointManager().addBreakpointListener(this);
+	}
+
+	@Override
+	public void dispose() {
+		DebugPlugin.getDefault().getBreakpointManager().removeBreakpointListener(this);
+		super.dispose();
+	}
+
+
+	@Override
+	public void run(IAction action) {
+		final IBreakpointManager breakpointManager = DebugPlugin.getDefault().getBreakpointManager();
+		final IBreakpoint[] breakpoints = breakpointManager.getBreakpoints();
+		if (breakpoints.length < 1) {
+			return;
+		}
+		IWorkbenchWindow window = DebugUIPlugin.getActiveWorkbenchWindow();
+		if (window == null) {
+			return;
+		}
+		IPreferenceStore store = DebugUIPlugin.getDefault().getPreferenceStore();
+		boolean prompt = store.getBoolean(IDebugPreferenceConstants.PREF_PROMPT_DISABLE_ALL_BREAKPOINTS);
+		boolean proceed = true;
+		if (prompt) {
+			MessageDialogWithToggle mdwt = MessageDialogWithToggle.openYesNoQuestion(window.getShell(),
+					this.fLabel1, this.fLabel2,
+					ActionMessages.AllBreakPointsActionEnablement_DontAskAgain, !prompt, null, null);
+			if (mdwt.getReturnCode() != IDialogConstants.YES_ID) {
+				proceed = false;
+			} else {
+				store.setValue(IDebugPreferenceConstants.PREF_PROMPT_DISABLE_ALL_BREAKPOINTS,
+						!mdwt.getToggleState());
+			}
+		}
+		if (proceed) {
+			new Job(this.fLabel2) {
+				@Override
+				protected IStatus run(IProgressMonitor monitor) {
+					try {
+						for (IBreakpoint breakpoint : breakpoints) {
+							breakpoint.setEnabled(fEnable);
+						}
+					} catch (CoreException e) {
+						DebugUIPlugin.log(e);
+						return Status.CANCEL_STATUS;
+					}
+					return Status.OK_STATUS;
```

Comment:
wouldn't it be more readable if "return Status.OK_STATUS;" was moved to the end of try block (between lines 115 and 116)?